src/set_theory/ordinal_arithmetic.lean

```diff
@@ -856,6 +857,25 @@ theorem sup_le {ι} {f : ι → ordinal} {a} : sup f ≤ a ↔ ∀ i, f i ≤ a
 theorem lt_sup {ι} {f : ι → ordinal} {a} : a < sup f ↔ ∃ i, a < f i :=
 by simpa only [not_forall, not_le] using not_congr (@sup_le _ f a)
 
+theorem sup_not_succ_of_lt_sup {ι} {f : ι → ordinal} (hf : ∀ i, f i < sup f) :
+  ∀ a < sup f, succ a < sup f :=
+begin
+  intros a hao,
+  by_contra' hoa,
+  have hao' := le_antisymm (succ_le.2 hao) hoa,
+  rw ←hao' at hf,
+  rw le_antisymm (le_of_lt hao) (sup_le.2 (λ i, lt_succ.1 (hf i))) at hao',
+  exact succ_ne_self _ hao',
+end
+
+-- A result that shows up twice for some reason.
+private lemma lt_sup_of_ne_sup {ι} {f : ι → ordinal} : (∀ i, f i ≠ sup f) → ∀ i, f i < sup f :=
+λ hf _, lt_of_le_of_ne (le_sup _ _) (hf _)
+
+theorem sup_not_succ_of_ne_sup {ι} {f : ι → ordinal} (hf : ∀ i, f i ≠ sup f) :
```

Comment:
This result is a stronger version of `sup_not_succ_of_lt_sup`, so you should just give this one and combine the proofs.

Reply:
Did just that in my new commit.